mudlet-mapper.xml

```diff
@@ -8063,6 +8064,7 @@ end</script>
 			mmp.settings:setOption(k, v, true)
 		end
 	end
+   mmp.oldcolour=mmp.settings.echocolour
```

Comment:
I'm not fond of the need to remember the old colour. Even less of storing that information in a public field.

Maybe we can extend the `createOption()` function to receive an optional `checkOption` function, that receives the old and new values and returns `true` if the new value is valid, otherwise `false`. This would be called in `setOption` before the assignment.

Reply:
I know it isn't the best solution, but unfortunately I'm not really comfortable messing with the options table code. I have a hard time wrapping my head around that code and the metatables and all that.